### PR TITLE
Feature/add mailer

### PR DESCRIPTION
**Pull request by bialikover**
Add Nodemailer and transport sendgrid.

### CODE REVIEW

File: getScreenShot.js [hunk 1]
```diff
@@ -8,18 +8,26 @@ module.exports = function gs(config) {
         getScreenShotList: getScreenShotList
     }
 
-    //get the list of screenshots in folder on a particular week.
-    function getScreenShotList() {
-        var path = config.screenShot.path;
+    function getScreenShotList(path) {
```
Comment (bialikover) [inline getScreenShot.js]: @chintan any thoughts on this?, 
just creating smaller functions

File: main.js [hunk 2]
```diff
@@ -1,13 +1,26 @@
 var config = require('./config.json');
 var gs = require('./getScreenShot')(config);
+var email = require('./email')(config);
+gs.getScreenShotList(config.screenShot.path)
```
Comment (bialikover) [inline main.js]: @chintan-patel Same here.. just creating named functions to handle promise responses.

File: email.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+module.exports = function email(config) {
+
+    return {
+        sendMail: sendMail
+    };
+
+    function sendMail(attachments, date) {
+        var email = {
+            to: config.mail.to,
+            from: config.mail.from,
```
Comment (chintan-patel) [inline email.js]: validation of email address required.....

File: email.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+            bcc: config.mail.bcc,
+            subject: config.mail.title + ' ' + buildDateTitle(date),
+            text: config.mail.body,
+            html: config.mail.html,
```
Comment (chintan-patel) [inline email.js]: validate html check?

File: email.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+            attachments: attachments
+        };
+
+        return mailer.sendMailAsync(email).then(function (res) {
+            return res;
+        });
```
Comment (chintan-patel) [inline email.js]: where is the catch block?

File: getScreenShot.js [hunk 6]
```diff
@@ -1,20 +1,32 @@
...
-    //get the list of screenshots in folder on a particular week.
-    function getScreenShotList() {
-        var path = config.screenShot.path;
-        var date = moment();
-        var year = date.year().toString();
-        var week = date.week().toString();
-        return readdir(path + '/' + year + '/' + week );
+    function getScreenShotList(p, date) {
+        var path = buildPath(p, date);
+        return readdir(path)
```
Comment (chintan-patel) [inline getScreenShot.js]: try to use function from `fs` npm module with `.readAsync` or something like that

File: main.js [hunk 7]
```diff
@@ -1,13 +1,28 @@
...
-    })
-    .finally(function () {
-        console.log(new Date());
-    });
+function sendMail(attachments) {
+    return email.sendMail(attachments, date);
+}
+
+function response(res) {
+    console.log(res);
```
Comment (chintan-patel) [inline main.js]: Add return statement saying `return res` to make the `promise chain` continue correctly.

File: main.js [hunk 8]
```diff
@@ -1,13 +1,28 @@
...
+}
+
+function handleError(e) {
+    console.log("There was an error: " + e);
```
Comment (chintan-patel) [inline main.js]: `return e`, even though its not used anywhere.

File: config.json [hunk 9]
```diff
@@ -1,12 +1,12 @@
...
+        "to": "bialikoer@gmail.com",
         "cc": "",
         "bcc": "",
         "title": "Weekly Time Sheet",
         "body": "Hello, \n\n Please find the time sheet attached.",
-        "path": ""
+        "html": "<h2>Weekly Time Sheet</h2><p>Hello, <br/> Please find the time sheet attached.</p>"
     },
     "screenShot": {
         "path": "/Users/adan/Desktop/Dropbox/Adan/docsUS/timeSheets/"
```
Comment (chintan-patel) [inline config.json]: Make it variable driven

**Pull request merged**